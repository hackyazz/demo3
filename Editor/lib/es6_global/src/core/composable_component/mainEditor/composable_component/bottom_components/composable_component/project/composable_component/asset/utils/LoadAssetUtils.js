

import * as Curry from "../../../../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as LogUtils$WonderEditor from "../../../../../../../../../utils/console/LogUtils.js";
import * as Caml_builtin_exceptions from "../../../../../../../../../../../../../node_modules/bs-platform/lib/es6/caml_builtin_exceptions.js";
import * as ConsoleUtils$WonderEditor from "../../../../../../../../../utils/ui/ConsoleUtils.js";
import * as FileNameService$WonderEditor from "../../../../../../../../../../service/atom/FileNameService.js";
import * as StateLogicService$WonderEditor from "../../../../../../../../../../service/stateTuple/logic/StateLogicService.js";

function getUploadAssetType(name) {
  var extname = FileNameService$WonderEditor.getExtName(name);
  switch (extname) {
    case ".glb" : 
        return /* LoadGLB */1;
    case ".jpeg" : 
    case ".jpg" : 
    case ".png" : 
        return /* LoadTexture */3;
    case ".wdb" : 
        return /* LoadWDB */0;
    case ".zip" : 
        return /* LoadGLTFZip */2;
    default:
      var partial_arg = LogUtils$WonderEditor.buildErrorMessage("the loaded asset type is error", "", "", "");
      StateLogicService$WonderEditor.getEditorState((function (param) {
              return ConsoleUtils$WonderEditor.error(partial_arg, param);
            }));
      return /* LoadError */5;
  }
}

function getUploadPackageType(name) {
  var extname = FileNameService$WonderEditor.getExtName(name);
  if (extname === ".wpk") {
    return /* LoadWPK */4;
  } else {
    var partial_arg = LogUtils$WonderEditor.buildErrorMessage("the loaded package type is error", "", "", "");
    StateLogicService$WonderEditor.getEditorState((function (param) {
            return ConsoleUtils$WonderEditor.error(partial_arg, param);
          }));
    return /* LoadError */5;
  }
}

function _handlePackageSpecificFuncByTypeSync(type_, handleWPKFunc) {
  if (type_ !== 4) {
    if (type_ >= 5) {
      return /* () */0;
    } else {
      throw [
            Caml_builtin_exceptions.match_failure,
            /* tuple */[
              "LoadAssetUtils.re",
              51,
              2
            ]
          ];
    }
  } else {
    return Curry._1(handleWPKFunc, /* () */0);
  }
}

function _handleAssetSpecificFuncByTypeSync(type_, param) {
  switch (type_) {
    case 0 : 
        return Curry._1(param[1], /* () */0);
    case 1 : 
        return Curry._1(param[2], /* () */0);
    case 2 : 
        return Curry._1(param[3], /* () */0);
    case 3 : 
        return Curry._1(param[0], /* () */0);
    case 4 : 
        throw [
              Caml_builtin_exceptions.match_failure,
              /* tuple */[
                "LoadAssetUtils.re",
                61,
                2
              ]
            ];
    case 5 : 
        return /* () */0;
    
  }
}

function readPakckageByTypeSync(reader, fileInfo) {
  return _handlePackageSpecificFuncByTypeSync(getUploadPackageType(fileInfo[/* name */0]), (function (param) {
                reader.readAsArrayBuffer(fileInfo[/* file */2]);
                return /* () */0;
              }));
}

function readAssetByTypeSync(reader, fileInfo) {
  return _handleAssetSpecificFuncByTypeSync(getUploadAssetType(fileInfo[/* name */0]), /* tuple */[
              (function (param) {
                  reader.readAsDataURL(fileInfo[/* file */2]);
                  return /* () */0;
                }),
              (function (param) {
                  reader.readAsArrayBuffer(fileInfo[/* file */2]);
                  return /* () */0;
                }),
              (function (param) {
                  reader.readAsArrayBuffer(fileInfo[/* file */2]);
                  return /* () */0;
                }),
              (function (param) {
                  reader.readAsArrayBuffer(fileInfo[/* file */2]);
                  return /* () */0;
                })
            ]);
}

export {
  getUploadAssetType ,
  getUploadPackageType ,
  _handlePackageSpecificFuncByTypeSync ,
  _handleAssetSpecificFuncByTypeSync ,
  readPakckageByTypeSync ,
  readAssetByTypeSync ,
  
}
/* LogUtils-WonderEditor Not a pure module */
