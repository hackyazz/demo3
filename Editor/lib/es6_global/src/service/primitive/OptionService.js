

import * as Curry from "../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Caml_option from "../../../../../node_modules/bs-platform/lib/es6/caml_option.js";
import * as OptionService$Wonderjs from "../../../../../node_modules/wonder.js/lib/es6_global/src/service/atom/OptionService.js";

function join(x) {
  if (x !== undefined) {
    return Caml_option.valFromOption(x);
  }
  
}

function andThenWithDefault(func, $$default, x) {
  if (x !== undefined) {
    return Curry._1(func, Caml_option.valFromOption(x));
  } else {
    return $$default;
  }
}

function either(someFunc, noneFunc, data, x) {
  if (x !== undefined) {
    return Curry._2(someFunc, data, Caml_option.valFromOption(x));
  } else {
    return Curry._1(noneFunc, data);
  }
}

function eitherWithNoData(someFunc, noneFunc, x) {
  if (x !== undefined) {
    return Curry._1(someFunc, Caml_option.valFromOption(x));
  } else {
    return Curry._1(noneFunc, /* () */0);
  }
}

function handleSomeAndIgnore(func, x) {
  if (x !== undefined) {
    Curry._1(func, Caml_option.valFromOption(x));
    return /* () */0;
  } else {
    return /* () */0;
  }
}

var unsafeGet = OptionService$Wonderjs.unsafeGet;

var unsafeGetJsonSerializedValue = OptionService$Wonderjs.unsafeGetJsonSerializedValue;

var isJsonSerializedValueNone = OptionService$Wonderjs.isJsonSerializedValueNone;

export {
  unsafeGet ,
  unsafeGetJsonSerializedValue ,
  isJsonSerializedValueNone ,
  join ,
  andThenWithDefault ,
  either ,
  eitherWithNoData ,
  handleSomeAndIgnore ,
  
}
/* OptionService-Wonderjs Not a pure module */
