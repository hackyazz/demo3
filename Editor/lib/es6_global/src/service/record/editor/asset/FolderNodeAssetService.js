

import * as Block from "../../../../../../../node_modules/bs-platform/lib/es6/block.js";
import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Caml_option from "../../../../../../../node_modules/bs-platform/lib/es6/caml_option.js";
import * as Log$WonderLog from "../../../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as LogUtils$WonderEditor from "../../../../core/utils/console/LogUtils.js";
import * as NodeAssetService$WonderEditor from "./NodeAssetService.js";
import * as UIStateAssetService$WonderEditor from "./UIStateAssetService.js";

function getNewFolderName(param) {
  return "New Folder";
}

function buildNode(nodeId, name, $staropt$star, param) {
  var children = $staropt$star !== undefined ? $staropt$star : UIStateAssetService$WonderEditor.build(undefined, undefined, /* () */0);
  return /* FolderNode */Block.__(3, [
            nodeId,
            /* record */[/* name */name],
            children
          ]);
}

function buildNodeByNodeData(nodeId, nodeData, children) {
  return /* FolderNode */Block.__(3, [
            nodeId,
            nodeData,
            children
          ]);
}

function _fatalShouldBeFolderNode(param) {
  return Log$WonderLog.fatal(LogUtils$WonderEditor.buildFatalMessage("should be folder node", "", "", ""));
}

function getNodeData(folderNode) {
  if (folderNode.tag === 3) {
    return folderNode[1];
  } else {
    return _fatalShouldBeFolderNode(/* () */0);
  }
}

function getChildren(folderNode) {
  if (folderNode.tag === 3) {
    return folderNode[2];
  } else {
    return _fatalShouldBeFolderNode(/* () */0);
  }
}

function getChildrenNodes(folderNode) {
  return UIStateAssetService$WonderEditor.get(getChildren(folderNode));
}

function rename(name, nodeData) {
  return /* record */[/* name */name];
}

function getNodeName(param) {
  return param[/* name */0];
}

function getIsShowChildren(folderNode) {
  if (folderNode.tag === 3) {
    return UIStateAssetService$WonderEditor.getIsShowChildrenByState(folderNode[2]);
  } else {
    return _fatalShouldBeFolderNode(/* () */0);
  }
}

function clearChildren(folderNode) {
  if (folderNode.tag === 3) {
    return /* FolderNode */Block.__(3, [
              folderNode[0],
              folderNode[1],
              UIStateAssetService$WonderEditor.map((function (param) {
                      return /* array */[];
                    }), folderNode[2])
            ]);
  } else {
    return _fatalShouldBeFolderNode(/* () */0);
  }
}

function filterChildrenById(targetNodeId, children) {
  return UIStateAssetService$WonderEditor.filter((function (prim, prim$1) {
                return prim$1.filter(Curry.__1(prim));
              }), (function (child) {
                return !NodeAssetService$WonderEditor.isIdEqual(NodeAssetService$WonderEditor.getNodeId(child), targetNodeId);
              }), children);
}

function findChild(folderNode, targetNode) {
  return UIStateAssetService$WonderEditor.find((function (prim, prim$1) {
                return Caml_option.undefined_to_opt(prim$1.find(Curry.__1(prim)));
              }), (function (childNode) {
                return NodeAssetService$WonderEditor.isNodeEqualById(childNode, targetNode);
              }), getChildren(folderNode));
}

function isFolderNode(node) {
  if (node.tag === 3) {
    return true;
  } else {
    return false;
  }
}

export {
  getNewFolderName ,
  buildNode ,
  buildNodeByNodeData ,
  _fatalShouldBeFolderNode ,
  getNodeData ,
  getChildren ,
  getChildrenNodes ,
  rename ,
  getNodeName ,
  getIsShowChildren ,
  clearChildren ,
  filterChildrenById ,
  findChild ,
  isFolderNode ,
  
}
/* Log-WonderLog Not a pure module */
