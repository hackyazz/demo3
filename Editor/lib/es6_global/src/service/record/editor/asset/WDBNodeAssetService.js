

import * as Block from "../../../../../../../node_modules/bs-platform/lib/es6/block.js";
import * as Log$WonderLog from "../../../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as LogUtils$WonderEditor from "../../../../core/utils/console/LogUtils.js";

function buildNode(nodeId, name, wdbGameObject) {
  return /* WDBNode */Block.__(2, [
            nodeId,
            /* record */[
              /* name */name,
              /* wdbGameObject */wdbGameObject
            ]
          ]);
}

function buildNodeByNodeData(nodeId, nodeData) {
  return /* WDBNode */Block.__(2, [
            nodeId,
            nodeData
          ]);
}

function getNodeData(node) {
  if (node.tag === 2) {
    return node[1];
  } else {
    return Log$WonderLog.fatal(LogUtils$WonderEditor.buildFatalMessage("should be wdb node", "", "", ""));
  }
}

function isWDBNode(node) {
  if (node.tag === 2) {
    return true;
  } else {
    return false;
  }
}

function rename(name, nodeData) {
  return /* record */[
          /* name */name,
          /* wdbGameObject */nodeData[/* wdbGameObject */1]
        ];
}

function getNodeName(param) {
  return param[/* name */0];
}

function getWDBGameObject(node) {
  return getNodeData(node)[/* wdbGameObject */1];
}

export {
  buildNode ,
  buildNodeByNodeData ,
  getNodeData ,
  isWDBNode ,
  rename ,
  getNodeName ,
  getWDBGameObject ,
  
}
/* Log-WonderLog Not a pure module */
