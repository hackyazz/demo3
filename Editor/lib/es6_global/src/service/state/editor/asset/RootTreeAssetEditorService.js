

import * as RootTreeAssetService$WonderEditor from "../../../record/editor/asset/RootTreeAssetService.js";
import * as TreeAssetEditorService$WonderEditor from "./TreeAssetEditorService.js";
import * as IndexAssetEditorService$WonderEditor from "./IndexAssetEditorService.js";

function getRootNode(editorState) {
  return RootTreeAssetService$WonderEditor.getRootNode(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function buildRootNode(name, editorState) {
  var match = RootTreeAssetService$WonderEditor.buildRootNode(RootTreeAssetService$WonderEditor.getAssetTreeRootName(/* () */0), IndexAssetEditorService$WonderEditor.getNodeIndex(editorState));
  return /* tuple */[
          match[0],
          match[1],
          IndexAssetEditorService$WonderEditor.setNodeIndex(match[2], editorState)
        ];
}

export {
  getRootNode ,
  buildRootNode ,
  
}
/* RootTreeAssetService-WonderEditor Not a pure module */
