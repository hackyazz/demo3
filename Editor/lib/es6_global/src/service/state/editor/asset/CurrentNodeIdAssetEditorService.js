

import * as NodeAssetService$WonderEditor from "../../../record/editor/asset/NodeAssetService.js";
import * as CurrentNodeIdAssetService$WonderEditor from "../../../record/editor/asset/CurrentNodeIdAssetService.js";
import * as RootTreeAssetEditorService$WonderEditor from "./RootTreeAssetEditorService.js";

function getCurrentNodeId(editorState) {
  return CurrentNodeIdAssetService$WonderEditor.getCurrentNodeId(editorState[/* assetRecord */2]);
}

function unsafeGetCurrentNodeId(editorState) {
  return CurrentNodeIdAssetService$WonderEditor.unsafeGetCurrentNodeId(editorState[/* assetRecord */2]);
}

function clearCurrentNodeId(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */CurrentNodeIdAssetService$WonderEditor.clearCurrentNodeId(editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function setCurrentNodeId(currentNodeId, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */CurrentNodeIdAssetService$WonderEditor.setCurrentNodeId(currentNodeId, editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function couldRemoveCurrentNode(editorState) {
  var match = CurrentNodeIdAssetService$WonderEditor.getCurrentNodeId(editorState[/* assetRecord */2]);
  if (match !== undefined) {
    return !NodeAssetService$WonderEditor.isIdEqual(match, NodeAssetService$WonderEditor.getNodeId(RootTreeAssetEditorService$WonderEditor.getRootNode(editorState)));
  } else {
    return false;
  }
}

export {
  getCurrentNodeId ,
  unsafeGetCurrentNodeId ,
  clearCurrentNodeId ,
  setCurrentNodeId ,
  couldRemoveCurrentNode ,
  
}
/* CurrentNodeIdAssetService-WonderEditor Not a pure module */
