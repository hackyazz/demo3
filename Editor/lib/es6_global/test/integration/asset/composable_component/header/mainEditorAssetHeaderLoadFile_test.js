

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as WDBTool$WonderEditor from "../../../../tool/WDBTool.js";
import * as LoadTool$WonderEditor from "../../tool/LoadTool.js";
import * as ConsoleTool$WonderEditor from "../../../../unit/tool/external/ConsoleTool.js";
import * as OptionService$WonderEditor from "../../../../../src/service/primitive/OptionService.js";
import * as ReactTestTool$WonderEditor from "../../../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../../../tool/GameObjectTool.js";
import * as LoadAssetUtils$WonderEditor from "../../../../../src/core/composable_component/mainEditor/composable_component/bottom_components/composable_component/project/composable_component/asset/utils/LoadAssetUtils.js";
import * as ImageDataMapTool$WonderEditor from "../../tool/ImageDataMapTool.js";
import * as EventListenerTool$WonderEditor from "../../../../unit/tool/EventListenerTool.js";
import * as BuildComponentTool$WonderEditor from "../../../../tool/BuildComponentTool.js";
import * as StateEditorService$WonderEditor from "../../../../../src/service/state/editor/StateEditorService.js";
import * as MainEditorAssetTool$WonderEditor from "../../tool/MainEditorAssetTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../tool/MainEditorSceneTool.js";
import * as WDBNodeAssetService$WonderEditor from "../../../../../src/service/record/editor/asset/WDBNodeAssetService.js";
import * as FolderNodeAssetService$WonderEditor from "../../../../../src/service/record/editor/asset/FolderNodeAssetService.js";
import * as MainEditorAssetNodeTool$WonderEditor from "../../tool/MainEditorAssetNodeTool.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../tool/MainEditorAssetTreeTool.js";
import * as MainEditorAssetUploadTool$WonderEditor from "../../tool/MainEditorAssetUploadTool.js";
import * as RootTreeAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/RootTreeAssetEditorService.js";
import * as OperateTreeAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/OperateTreeAssetEditorService.js";
import * as CurrentNodeIdAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/CurrentNodeIdAssetEditorService.js";
import * as MainEditorAssetChildrenNodeTool$WonderEditor from "../../tool/MainEditorAssetChildrenNodeTool.js";
import * as SelectedFolderNodeIdInAssetTreeAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/SelectedFolderNodeIdInAssetTreeAssetEditorService.js";

describe("MainEditorAssetHeader->load file", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var cubeTexturedWDBArrayBuffer = /* record */[/* contents */1];
        beforeAll((function () {
                cubeTexturedWDBArrayBuffer[0] = WDBTool$WonderEditor.convertGLBToWDB("CubeTextured");
                return /* () */0;
              }));
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, (function () {
                        return /* () */0;
                      }));
                return Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
              }));
        afterEach((function () {
                Curry._1(Sinon.restoreSandbox, sandbox[0]);
                StateEditorService$WonderEditor.setState(SelectedFolderNodeIdInAssetTreeAssetEditorService$WonderEditor.clearSelectedFolderNodeIdInAssetTree(CurrentNodeIdAssetEditorService$WonderEditor.clearCurrentNodeId(StateEditorService$WonderEditor.getState(/* () */0))));
                return /* () */0;
              }));
        describe("test load file", (function () {
                beforeEach((function () {
                        Curry._1(MainEditorAssetTool$WonderEditor.buildFakeFileReader, /* () */0);
                        return Curry._1(MainEditorAssetTool$WonderEditor.buildFakeImage, /* () */0);
                      }));
                describe("test snapshot", (function () {
                        describe("if not select specific treeNode", (function () {
                                return Wonder_jest.testPromise("load file should add into root node children", (function () {
                                              MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                              return MainEditorAssetUploadTool$WonderEditor.loadOneTexture(undefined, undefined, undefined, undefined, /* () */0).then((function () {
                                                            return Promise.resolve(ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0)));
                                                          }));
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("test logic", (function () {
                        describe("test should add into root node children", (function () {
                                return Wonder_jest.testPromise("test children node length", (function () {
                                              MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                              var originChildrenLen = FolderNodeAssetService$WonderEditor.getChildrenNodes(RootTreeAssetEditorService$WonderEditor.getRootNode(StateEditorService$WonderEditor.getState(/* () */0))).length;
                                              return MainEditorAssetUploadTool$WonderEditor.loadOneTexture(undefined, undefined, undefined, undefined, /* () */0).then((function () {
                                                            return Promise.resolve(Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](FolderNodeAssetService$WonderEditor.getChildrenNodes(RootTreeAssetEditorService$WonderEditor.getRootNode(StateEditorService$WonderEditor.getState(/* () */0))).length - originChildrenLen | 0), 1));
                                                          }));
                                            }));
                              }));
                        describe("test should add into nodeMap", (function () {
                                describe("test imageNodeMap", (function () {
                                        Wonder_jest.testPromise("add image base64 to imageNodeMap", (function () {
                                                MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                                var imgBase64 = "newImgBase64";
                                                return MainEditorAssetUploadTool$WonderEditor.loadOneTexture(undefined, undefined, undefined, imgBase64, /* () */0).then((function (uploadedTextureNodeId) {
                                                              MainEditorAssetChildrenNodeTool$WonderEditor.selectTextureNode(uploadedTextureNodeId, undefined, undefined, /* () */0);
                                                              var editorState = StateEditorService$WonderEditor.getState(/* () */0);
                                                              var match = ImageDataMapTool$WonderEditor.getDataByTextureNode(MainEditorAssetNodeTool$WonderEditor.unsafeGetCurrentNode(editorState), editorState);
                                                              return Promise.resolve(Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](OptionService$WonderEditor.unsafeGet(match[/* base64 */0])), imgBase64));
                                                            }));
                                              }));
                                        return Wonder_jest.testPromise("test show texture image, get it base64 from imageNodeMap", (function () {
                                                      MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                                      return MainEditorAssetUploadTool$WonderEditor.loadOneTexture(undefined, undefined, undefined, "newImgBase64", /* () */0).then((function () {
                                                                    return Promise.resolve(ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0)));
                                                                  }));
                                                    }));
                                      }));
                                describe("test textureNodeMap", (function () {
                                        return Wonder_jest.testPromise("add created texture index to textureNodeMap", (function () {
                                                      MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                                      return MainEditorAssetUploadTool$WonderEditor.loadOneTexture(undefined, undefined, undefined, undefined, /* () */0).then((function (uploadedTextureNodeId) {
                                                                    MainEditorAssetChildrenNodeTool$WonderEditor.selectTextureNode(uploadedTextureNodeId, undefined, undefined, /* () */0);
                                                                    return Promise.resolve(Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](MainEditorAssetNodeTool$WonderEditor.getTextureComponentFromCurrentNodeId(/* () */0)), 0));
                                                                  }));
                                                    }));
                                      }));
                                describe("test wdbNodeMap", (function () {
                                        beforeEach((function () {
                                                LoadTool$WonderEditor.buildFakeTextDecoder(LoadTool$WonderEditor.convertUint8ArrayToBuffer);
                                                LoadTool$WonderEditor.buildFakeURL(sandbox[0]);
                                                return LoadTool$WonderEditor.buildFakeLoadImage();
                                              }));
                                        return Wonder_jest.testPromise("add name, wdbGameObject to wdbNodeMap", (function () {
                                                      MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                                      var fileName = "CubeTextured";
                                                      var newGameObjectUid = GameObjectTool$WonderEditor.getNewGameObjectUid(undefined, /* () */0);
                                                      return MainEditorAssetUploadTool$WonderEditor.loadOneWDB(cubeTexturedWDBArrayBuffer[0], undefined, undefined, fileName, /* () */0).then((function (uploadedWDBNodeId) {
                                                                    var editorState = StateEditorService$WonderEditor.getState(/* () */0);
                                                                    var match = WDBNodeAssetService$WonderEditor.getNodeData(OperateTreeAssetEditorService$WonderEditor.unsafeFindNodeById(uploadedWDBNodeId, editorState));
                                                                    return Promise.resolve(Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                                                        match[/* name */0],
                                                                                        match[/* wdbGameObject */1]
                                                                                      ]), /* tuple */[
                                                                                    fileName,
                                                                                    newGameObjectUid
                                                                                  ]));
                                                                  }));
                                                    }));
                                      }));
                                return /* () */0;
                              }));
                        return /* () */0;
                      }));
                describe("deal with specific case", (function () {
                        return Wonder_jest.test("if upload error file type, should error", (function () {
                                      ConsoleTool$WonderEditor.notShowMessage(/* () */0);
                                      BuildComponentTool$WonderEditor.buildConsole(undefined, undefined, /* () */0);
                                      var errorStub = Curry._3(Sinon.createMethodStub, sandbox[0], console, "error");
                                      LoadAssetUtils$WonderEditor._handleAssetSpecificFuncByTypeSync(LoadAssetUtils$WonderEditor.getUploadAssetType("aaa.bb"), /* tuple */[
                                            (function () {
                                                return /* () */0;
                                              }),
                                            (function () {
                                                return /* () */0;
                                              }),
                                            (function () {
                                                return /* () */0;
                                              }),
                                            (function () {
                                                return /* () */0;
                                              })
                                          ]);
                                      return ConsoleTool$WonderEditor.judgeError("type is error", errorStub);
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
