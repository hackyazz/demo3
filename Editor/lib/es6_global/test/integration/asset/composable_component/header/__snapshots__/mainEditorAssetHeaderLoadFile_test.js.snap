// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorAssetHeader->load file test load file test logic test should add into nodeMap test imageNodeMap test show texture image, get it base64 from imageNodeMap 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
            onMouseDown={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container select-not-active"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-fileBox "
      onClick={[Function]}
    >
      <div
        className="box-image"
      >
        <img
          onDragStart={[Function]}
          src="newImgBase64"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          loadImg
        </span>
      </div>
    </article>
  </article>
</article>
`;

exports[`MainEditorAssetHeader->load file test load file test snapshot if not select specific treeNode load file should add into root node children 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
            onMouseDown={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container select-not-active"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-fileBox "
      onClick={[Function]}
    >
      <div
        className="box-image"
      >
        <img
          onDragStart={[Function]}
          src="newImgBase64"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          loadImg
        </span>
      </div>
    </article>
  </article>
</article>
`;
